src/gt4py/next/program_processors/runners/roundtrip.py

```diff
@@ -151,7 +159,9 @@ def fencil_generator(
         """
     )
 
-    with tempfile.NamedTemporaryFile(mode="w", suffix=".py", delete=False) as source_file:
+    with tempfile.NamedTemporaryFile(
```

Comment:
Not sure what you mean, but note that I added `encoding="utf-8"` for a problem that I don't fully understand (because to my understanding it should have defaulted to utf-8 anyway), but fixed a problem in a test that contained non-ascii characters.